### ISSUE TITLE
ProhibitCaptureWithoutTest does not understand indented heredocs

### ISSUE THREAD

**Issue by PatrickCronin** [Perl-Critic/Perl-Critic#845]
```
$perlcritic -version
1.132
```

Code:
```
    my $api_response = $dbh->selectrow_arrayref(<<~'_SQL', {}, $bind_param )->[0];
            SELECT COUNT(*)
            FROM foo
            WHERE bar = $1
            _SQL
```
Expected: `perlcritic -3` realizes that that `$1` is included in an uninterpolated intended heredoc, and produces no error.
Actual: `perlcritic -3` produces a `RegularExpressions::ProhibitCaptureWithoutTest` warning.

If we unindent the heredoc, as in:
```
    my $api_response = $dbh->selectrow_arrayref(<<'_SQL', {}, $bind_param )->[0];
SELECT COUNT(*)
FROM foo
WHERE bar = $1
_SQL
```
then `perlcritic -3` does not report a warning.

**Comment by petdance** [comment]
My first guess is that PPI doesn't handle indented heredocs yet, but I haven't followed PPI dev in a while.

**Comment by oschwald** [comment]
Your guess is correct. [PPI doesn't handle it yet](https://github.com/adamkennedy/PPI/pull/209).

**Comment by petdance** [comment]
Excellent.  Thanks for the research, and a link we can now keep an eye on.

**Comment by wchristian** [comment]
PPI should support this behavior now: https://metacpan.org/release/MITHALDU/PPI-1.246

**Comment by petdance** [comment]
Merged as 8b384b4.  Thank you!
